src/libsyntax/attr/builtin.rs

```diff
@@ -665,7 +665,18 @@ fn find_deprecation_generic<'a, I>(sess: &ParseSess,
             break
         }
 
-        let meta = attr.meta().unwrap();
+        let meta = match attr.meta() {
+            Some(meta) => meta,
+            None => {
+                // `#[deprecated]` does not follow meta grammar,
+                // but it should according to the spec in `BUILTIN_ATTRIBUTES`.
+                // We have previously emitted an error in `validate_attr::check_meta`.
+                diagnostic.delay_span_bug(attr.span, "non-meta grammar in deprecated attribute");
+                // But we can at least conclude that deprecation is desired.
+                depr = Some(Deprecation { since: None, note: None });
+                continue;
+            }
+        };
```

Comment:
```suggestion
        let meta = match attr.meta() {
            Some(meta) => meta,
            None => continue,
        };
```
In this case anything besides that is either noise or overthinking, IMO.

`meta()` and its derivatives are used in numerous places which just skip the attribute in case of `None` and without using `delay_span_bug`.
`deprecated` is primarily targeted at other crates which will never see it in case of the attribute format error.